src/types.rs

```diff
@@ -245,98 +257,173 @@ impl Default for DampingFactors {
 }
 
 #[derive(Debug)]
-pub enum Dependency {
-    Contrib(Weight),
-    ContribPrime(Weight),
-    Maintain(Weight),
-    MaintainPrime(Weight),
-    Depend(Weight),
+pub struct Dependency<Id, W> {
+    id: Id,
+    dependency_type: DependencyType<W>,
 }
 
-impl fmt::Display for Dependency {
+impl<Id, W> fmt::Display for Dependency<Id, W>
```

Comment:
Nothing urgent, but I find the wording of `Dependency` misleading, as there is also the `Depend` type. Maybe we could rename this to something more generic.

Reply:
Yeah, agreed. I have tried in the beginning to use the same wording as the paper, but I suck at names 😁 Let's brainstorm on what might be more effective! `Relationship`? 🤔